core/adaptive/config.go

```diff
@@ -0,0 +1,94 @@
+// Copyright 1999-2020 Alibaba Group Holding Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package adaptive
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/alibaba/sentinel-golang/util"
+)
+
+// AdaptiveType indicates the adaptive type.
+type AdaptiveType int32
```

Comment:
AdaptiveType and CalculateStrategy should support to register.
You can refer to PR: https://github.com/alibaba/sentinel-golang/pull/386

Reply:
done